CBL-6452 CBL-6433 CBL-6457: New logging API

borrrden opened the pull request:
Each ticket is handled in a separate commit, with details on the relevant commit

borrrden commented:
The test failure appears unrelated to this area (timeout on a P2P test)

bmeike reviewed (commented):
I'm going to need more time to read this...

borrrden commented on src/Couchbase.Lite.Shared/Support/Android/AndroidConsoleLogger.cs:
```diff
@@ -32,7 +32,7 @@ internal sealed class AndroidConsoleLogger : IConsoleLogger
         public LogDomain Domains { get; set; } = LogDomain.All;
 
-        public LogLevel Level { get; set; } = LogLevel.Warning;
+        public LogLevel Level { get; set; } = LogLevel.None;
```
Because otherwise the logging is going to be in two places right?  I disabled the old api by default so that the new one could be used.  I don’t see any benefit in keeping the old one defaulted to warning when the new one has a default of warning.

borrrden commented on src/Couchbase.Lite.Shared/Log/LogTo.cs:
```diff
@@ -125,6 +125,34 @@ internal void W(string tag, string format, params object?[] args)
+        internal static bool OldApiUsed()
+        {
+#pragma warning disable CS0618 // Type or member is obsolete
+            return Database.Log.Console.Level != LogLevel.None
```
Because the new default is None for me.

borrrden commented:
@pasin I changed back to having the default be warning for the old config and updated my "old api used" detection to check for deviation from that.  The net result is as follows:

1. No Change
    Console warning log level via new API
2. Modify anything on the new API (e.g. add file logger)
    New API used, old API modifications are now an error, console log level still warning via new API
3. Modify anything on the old API (e.g. Add file logging config)
    Old API used, new API modifications are now an error, console log level still warning via old API
